tests/e2e/tests/content-manager/history.spec.ts

```diff
@@ -218,14 +218,25 @@ describeOnCondition(edition === 'EE')('History', () => {
       await page.getByRole('combobox', { name: 'Authors' }).click();
       await page.getByText('Will Kitman').click();
       await page.getByRole('combobox', { name: 'Authors' }).click();
+
+      // Wait for the relation response before saving, otherwise the document might save without the added relation
+      const relationResponsePromise = page.waitForResponse(async (response) => {
+        return (
+          response.status() === 200 &&
+          response.url().includes('content-manager/relations/api::article.article/authors')
+        );
+      });
       await page.getByText('Coach Beard').click();
+      await relationResponsePromise;
+
       await page.getByRole('button', { name: 'Save' }).click();
-      await page.waitForURL(ARTICLE_EDIT_URL);
+      // Confirm the save was succesful before proceeding, otherwise we may end up on the related page before the relation is established
+      await findAndClose(page, 'Saved Document');
 
       // Delete one of the authors, leaving only Coach Beard
       await clickAndWait(page, page.getByRole('link', { name: 'Will Kitman' }));
       await page.waitForURL(AUTHOR_EDIT_URL);
-      await page.getByRole('button', { name: /more actions/i }).click();
+      await page.getByRole('button', { name: 'More actions' }).click();
```

Comment:
I don't think this change is necessary but I'm not too bothered by unless y'all are